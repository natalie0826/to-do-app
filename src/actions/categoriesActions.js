export const categoriesActions = {
    ADD_CATEGORY: 'ADD_CATEGORY',
    FETCH_CATEGORIES: 'FETCH_CATEGORIES',
    FETCH_CATEGORIES_SUCCESS: 'FETCH_CATEGORIES_SUCCESS',
    FETCH_CATEGORIES_FAILURE: 'FETCH_CATEGORIES_FAILURE',
}