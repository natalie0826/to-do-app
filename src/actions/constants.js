export const constants = {
    ADD_TODO: 'ADD_TODO',
    EDIT_TODO: 'EDIT_TODO',
    DELETE_TODO: 'DELETE_TODO',
    TOGGLE_TODO: 'TOGGLE_TODO',
    REQUEST_TODOS: 'REQUEST_TODOS',
    RECEIVE_TODOS: 'RECEIVE_TODOS',
    SET_FILTER: 'SET_FILTER',
    ADD_CATEGORY: 'ADD_CATEGORY',

    FETCH_TODOS: 'FETCH_TODOS',
    FETCH_TODOS_SUCCESS: 'FETCH_TODOS_SUCCESS',
    FETCH_TODOS_FAILURE: 'FETCH_TODOS_FAILURE',
};
