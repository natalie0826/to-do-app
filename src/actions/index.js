// actions call reducers

let nextId = 0;

export const addTodo = (text) => ({
    type: 'ADD_TODO',
    id: nextId++,
    text
});

export const editTodo = (id, text) => ({
    type: 'EDIT_TODO',
    id,
    text
});

export const deleteTodo = (id) => ({
    type: 'DELETE_TODO',
    id
});

export const toggleTodo = (id) => ({
    type: 'TOGGLE_TODO',
    id
});

export const restoreTodos = () => ({
    type: 'RESTORE_DELETED'
});

export const setFilter = (filter) => ({
    type: 'SET_FILTER',
    filter
});
