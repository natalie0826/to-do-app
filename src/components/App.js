import React from 'react';
import ShowTodoList from '../containers/ShowTodoList';
import AddTodo from '../containers/AddTodo';
import FilterBlock from '../components/FilterBlock';

const App = () => (
    <div>
        <AddTodo />
        <ShowTodoList />
        <FilterBlock />
    </div>
)

export default App;
