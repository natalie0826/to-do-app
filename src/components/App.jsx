import React from 'react';
import ShowTodoList from '../containers/ShowTodoList';
import AddTodo from '../containers/AddTodo';
import FilterBlock from '../components/FilterBlock';
import Composition from './Composition';

export default class App extends React.Component {
    render() {
        return (
            <Composition header="Welcome to the To-Do!"
                left={
                    <div>
                        <AddTodo />
                        <ShowTodoList />
                    </div>
                }
                right={
                    <FilterBlock />
                }
            />
        )
    }
}
