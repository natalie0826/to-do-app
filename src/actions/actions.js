// actions call reducers
import fetch from 'isomorphic-fetch';
import uuidv4 from 'uuid/v4';

import { constants } from './constants';

export const addTodo = (text, category, description, completed, deleted) => ({
    type: constants.ADD_TODO,
    id: uuidv4(),
    text,
    category,
    description,
    completed,
    deleted
});

export const editTodo = (id, text, category, description) => ({
    type: constants.EDIT_TODO,
    id,
    text,
    category,
    description
});

export const deleteTodo = (id) => ({
    type: constants.DELETE_TODO,
    id
});

export const toggleTodo = (id, completed) => ({
    type: constants.TOGGLE_TODO,
    id,
    completed
});

export const requestTodos = () => ({
    type: constants.REQUEST_TODOS
});

export const fetchTodos = () => {
    return (dispatch) => {
        return fetch('https://api.myjson.com/bins/aufxd')
            .then(response => response.json())
            .then(todos => todos.map(todo => dispatch(addTodo(todo.text, todo.category, todo.text, todo.completed, todo.deleted)))); // addTodo = initializeStore
    };
};

export const setFilter = (filter) => ({
    type: constants.SET_FILTER,
    filter
});

export const addCategory = (category, color) => ({
    type: constants.ADD_CATEGORY,
    category,
    color
});