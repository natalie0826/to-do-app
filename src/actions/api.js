import axios from 'axios';
import { urls } from '../constants/urls';

export default axios.create({
    baseUrl: urls.todos
});
