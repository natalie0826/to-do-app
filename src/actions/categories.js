import { constants } from './constants';

export const addCategory = (category) => ({
    type: constants.ADD_CATEGORY,
    payload: {
        category: category.category,
        color: category.color
    }
});

export const fetchCategories = (url) => {
    return (dispatch) => {
        return fetch(url)
            .then(response => response.json())
            .then(categories => categories.map(category => dispatch(addCategory(category.category, category.color))));
    };
};
