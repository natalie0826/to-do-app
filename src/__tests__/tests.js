import React from 'react';
import { shallow } from 'enzyme';

// components
import App from '../components/App';

// actions
import * as actions from '../actions/index';

describe('App component', () => {
    const wrapper = shallow(<App />);

    it('renders title', () => {
        const welcome = <h3>Todos</h3>;
        // expect(wrapper.contains(welcome)).to.equal(true);
        // expect(wrapper.contains(welcome)).toEqual(true);
        expect(wrapper).toContainReact(welcome);
    });
});

describe('Actions', () => {
    const id = 0;
    const text = 'test';

    it('should create an action to ADD a todo', () => {
        const date = new Date();
        const expectedAction = {
            type: 'ADD_TODO',
            text,
            date,
            id
        };
        expect(actions.addTodo(text, date)).toEqual(expectedAction);
    });

    it('should create an action to DELETE the todo', () => {
        const expectedAction = {
            type: 'DELETE_TODO',
            id
        };
        expect(actions.deleteTodo(id)).toEqual(expectedAction);
    });

    it('should create an action to EDIT the todo', () => {
        const expectedAction = {
            type: 'EDIT_TODO',
            text,
            id
        };
        expect(actions.editTodo(id, text)).toEqual(expectedAction);
    });
})
