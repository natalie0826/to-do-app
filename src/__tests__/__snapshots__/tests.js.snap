// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application render todo should render correctly 1`] = `
<Connect(Todo)
  completed={false}
  deleted={false}
  id={2}
  key="1"
  text="aaaa"
/>
`;
